src/io/flutter/actions/DeviceSelectorAction.java

```diff
@@ -92,12 +92,16 @@ private boolean isSelectorVisible(@Nullable Project project) {
   private void updateActions(@NotNull Project project, Presentation presentation) {
     actions.clear();
 
-    for (FlutterDevice item : DeviceService.getInstance(project).getConnectedDevices()) {
+    final DeviceService service = DeviceService.getInstance(project);
+
+    for (FlutterDevice item : service.getConnectedDevices()) {
       actions.add(new SelectDeviceAction(item));
     }
 
     if (actions.isEmpty()) {
-      actions.add(new NoDevicesAction());
+      final boolean isLoading = service.getStatus() == DeviceService.State.LOADING;
+      final String message = isLoading ? "Loading..." : "No devices";
```

Comment:
Consider externalizing these Strings?  See calls to `FlutterBundle.message(..)` for examples...